core/src/test/scala/org/apache/spark/sql/store/TokenizationTest.scala

```diff
@@ -0,0 +1,252 @@
+      val cacheMap = SnappySession.getPlanCache.asMap()
+      assert( cacheMap.size() == 1)
+      val x = cacheMap.keySet().toArray()(0).asInstanceOf[CachedKey].sqlText
+      assert(x.equals(q(0)))
+      snc.sql(s"drop table $all_typetable")
+    } finally {
+      snc.sql("set spark.sql.caseSensitive = false")
+      snc.sql("set schema = APP")
+    }
+
+    logInfo("Successful")
+  }
+
+  test("Test tokenize for joins and sub-queries") {
+    val numRows = 10
+    createSimpleTableAndPoupulateData(numRows, s"$table", true)
+    createSimpleTableAndPoupulateData(numRows, s"$table2")
+    var query = s"select * from $table t1, $table2 t2 where t1.a in " +
+      s"( select a from $table2 where b = 5 )"
+    snc.sql(query).collect()
+
+    val cacheMap = SnappySession.getPlanCache.asMap()
+
+    assert( cacheMap.size() == 1)
+
+    query = s"select * from $table t1, $table2 t2 where t1.a in " +
+      s"( select a from $table2 where b = 100 )"
+    snc.sql(query).collect()
+    assert( cacheMap.size() == 1)
+    logInfo("Successful")
+  }
+
+  test("Test tokenize for nulls") {
+    logInfo("Successful")
+  }
+
+  test("Test tokenize for cast queries") {
+    logInfo("Successful")
+  }
+
+  private def createSimpleTableAndPoupulateData(numRows: Int, name: String, dosleep: Boolean = false) = {
+    val data = ((0 to numRows), (0 to numRows), (0 to numRows)).zipped.toArray
+    val rdd = sc.parallelize(data, data.length)
+      .map(s => Data(s._1, s._2, s._3))
+    val dataDF = snc.createDataFrame(rdd)
+
+    snc.sql(s"Drop Table if exists $name")
+    snc.sql(s"Create Table $name (a INT, b INT, c INT) " +
+      "using column options()")
+    dataDF.write.insertInto(s"$name")
+    // This sleep was necessary as it has some dependency on the region size
+    // collector thread frequency. Can't remember right now.
+    if (dosleep) Thread.sleep(5000)
+  }
+
+  private def createAllTypeTableAndPoupulateData(numRows: Int, name: String, dosleep: Boolean = false) = {
+    val ints = (0 to numRows).zipWithIndex.map {case (_, i) =>
+      i
+    }
+    val longs = (0 to numRows).zipWithIndex.map    { case (_, i) => 1L*1000*i }
+    val floats = (0 to numRows).zipWithIndex.map   { case (_, i) => 0.1f*i    }
+    val decimals = (0 to numRows).zipWithIndex.map { case (_, i) => 0.1d*i    }
+    val strs = (0 to numRows).zipWithIndex.map     { case (_, i) => s"abc$i"  }
+    val dates = (0 to numRows).zipWithIndex.map    { case (_, i) => 1         }
+    val tstmps = (0 to numRows).zipWithIndex.map   { case (_, i) => s"abc$i"  }
+
+    val x =((((((ints, longs).zipped.toArray, floats).zipped.toArray,
+      decimals).zipped.toArray, strs).zipped.toArray, dates).zipped.toArray, tstmps).zipped.toArray
+    val data = x map { case ((((((i, l), f), d), s), dt), ts) =>
+      (i, l, f, d, s, dt, ts)
+    }
+
+    val rdd = sc.parallelize(data, data.length)
+      .map(s => (s._1, s._2, s._3, s._4, s._5, s._6, s._7))
+    val dataDF = snc.createDataFrame(rdd)
+
```

Comment:
I think the test is not verifying tokenization of all data types because date is represented as Int ? ts is represented as Long ? Also, the code can be much simpler like 
```java
// code is compatible with jdk8 & above
    val data = (0 to numRows).zipWithIndex.map { case (_, i) =>
      val ins = Instant.now()
      (i, 1L * 1000 * i, 0.1f * i, 0.1d * i, s"abc$i",
          ins.toEpochMilli + i, ins.toString)
    }

    val rdd = sc.parallelize(data, data.length)
      .map(s => (s._1, s._2, s._3, s._4, s._5, s._6, s._7))
    val dataDF = snc.createDataFrame(rdd)

    snc.sql(s"Drop Table if exists $name")
    snc.sql(s"Create Table $name (a INT, b Long, c Float, d Double, s String, dt DATE, ts TIMESTAMP)"
        + "using column options()")
```
note: dt and ts types in the table is changed and insert ops is expected to auto typecast the input values.
I haven't considered timezone implications in the above code.